### ISSUE TITLE
[kinozal] (testing) Test search in Kinozal => Found no results while trying to browse this tracker

### ISSUE THREAD

**Issue by dshkoda32** [Jackett/Jackett#13343]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Starting Jackett Version v0.20.1281
OS version: Unix 5.4.17.2102 (64bit OS) (64bit process)
Jackett variant: CoreLinuxArm64
File /etc/issue: Ubuntu 20.04.4 LTS \n \l
Running in Docker: Yes (image build: v0.20.1197-ls70)
ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads
App config/log directory: /config/Jackett
Using proxy: Disabled
Using FlareSolverr: Disabled
```

### Description

When I enter the official link kinozal.tv , then jackett in the movie viewing application does not give a hash when entering a mirror https://kinozal-tv.appspot.com / does not pass the test.

### Logged Error Messages

```text
...&cat=2030,2040&q=big+buck+bunny will search for "big buck bunny" in the Movies/SD (2030) and Movies/HD (2040) categories (See the indexer configuration for available categories).
Jackett Configuration
  

Admin password: 
••••••••••
  
Base path override: 
/jackett
Base URL override: 
http://jackett:9117
Server port: 
9117
Blackhole directory: 
c:\torrents\
Proxy type: 
Disabled
Proxy URL: 
Proxy port: 
Proxy username: 
Proxy password: 
External access: 
Allow CORS: 
Disable auto update: 
Update to pre-release: 
Enhanced logging: 
Cache enabled (recommended): 
Cache TTL (seconds): 
2100
Cache max results per indexer: 
1000
FlareSolverr API URL: 
Blank for default
FlareSolverr Max Timeout (ms): 
55000
OMDB API key: 
OMDB API Url: 
Blank for default
Jackett Version v0.20.1281
×
Server Logs
Date	Level	Message
2022-07-06 11:16:08	Error	
System.Exception: Test search in Kinozal => Found no results while trying to browse this tracker
   at Jackett.Common.Services.IndexerManagerService.TestIndexer(String name) in /home/vsts/work/1/s/src/Jackett.Common/Services/IndexerManagerService.cs:line 314
   at Jackett.Server.Controllers.IndexerApiController.Test() in /home/vsts/work/1/s/src/Jackett.Server/Controllers/IndexerApiController.cs:line 132
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/vsts/work/1/s/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 61
```

**Comment by garfield69** [comment]
>When I enter the official link kinozal.tv , then jackett in the movie viewing application does not give a hash when entering a mirror https://kinozal-tv.appspot.com/ / does not pass the test.

The Jackett kinozal indexer only supports the use of the default sitelink.
while the appspot domain may or may not be a official mirror, when you use this domain and the indexer attempts to perform a browse with a http get to https://kinozal-tv.appspot.com/browse.php?c=0&s=&g=0&v=0&d=0&w=0&t=0&f=0 that site returns a redirect 
```
https://kinozal-tv.appspot.com/browse.php?sid=wrbHbA08&c=0&s=&g=0&v=0&d=0&w=0&t=0&f=0 185 bytes
<a href="https://kinozal-tv.appspot.com/browse.php?sid=wrbHbA08&c=0&s=&g=0&v=0&d=0&w=0&t=0&f=0">https://kinozal-tv.appspot.com/browse.php?sid=wrbHbA08&c=0&s=&g=0&v=0&d=0&w=0&t=0&f=0</a>
```
which is not the HTML page that contains the results of the search.
Therefore, you cannot use that mirror with the Jackett kinozal indexer.
Basically, this mirror is not a complete  one-for-one replacement for the official domain , it is just different enough that it will not work. Something to do with the sid= parm that the mirror is inserting in the get request, which prevents the indexer from working. 
We have no plans to build a new indexer just for this mirror, it would require one coded in C# and C# devs are very scarce ATM.

**Comment by garfield69** [comment]
Changed my mind.
Did some experiments with adding followredirect to the kinozal indexer, and added some amendments to some of the selectors (name, details, download) to cope with the slightly different links from the original, and it seems to be working.
I'll post an updated indexer shortly after some more testing, and the next release of Jackett will include the updated indexer if all goes to plan.

**Comment by dshkoda32** [comment]
Thank you

**Comment by garfield69** [comment]
v0.20.1290

**Comment by dshkoda32** [comment]
Everything works fine, thank you so much

**Comment by dshkoda32** [comment]
Please add this address to the kinozal indexer https://kinozal-guru.appspot.com /, and then the rest are blocked and writes during the test that there are no results

**Comment by ilike2burnthing** [comment]
http://kinozal.tv/ is working here.

As far as I can tell http://kinozal.guru/ is a completely different tracker. Accounts for *.tv do not work on *.guru.

**Comment by dshkoda32** [comment]
Accounts from TV are suitable for guru. Because of the locks, I can't log in to tv. Guru opens without problems with the same account

**Comment by ilike2burnthing** [comment]
Apologies, you're correct, I was typing in the wrong password. I'll add those now and they'll be available in the next release in about ~11hrs. You can just type them into the site link field now though.

**Comment by dshkoda32** [comment]
Thanks you

**Comment by dshkoda32** [comment]
Thank you for adding the address, but it's a pity that the indexer test does not come to it, although I go to this address from the browser without any locks

**Comment by ilike2burnthing** [comment]
Well that's what I get for not testing the appspot mirrors - the tv one appears to be dead or down, and the guru one is trying re-login upon search (possibly something to do with it want an `sid`).

I'll see if I can resolve this, but if not I'll just have to remove it, and you'll just have to use a proxy of VPN if you're blocked.

**Comment by dshkoda32** [comment]
Thank you very much, everything works. You are just a wizard))

**Comment by garfield69** [comment]
v0.20.1575

**Comment by dshkoda32** [comment]
The tracker began to issue an error "Login Failed, got redirected."

**Comment by ilike2burnthing** [comment]
Working ok here. Reconfigure the indexer, and check if you need to update the SID.

**Comment by dshkoda32** [comment]
The password with the login is correct, the sid has been changed to a new one, the site is accessed without problems from the browser, but this error is still in the indexer

**Comment by dshkoda32** [comment]
ted.
   at Jackett.Common.Indexers.CardigannIndexer.TestLogin() in /home/vsts/work/1/s/src/Jackett.Common/Indexers/CardigannIndexer.cs:line 834
   at Jackett.Common.Indexers.CardigannIndexer.ApplyConfiguration(JToken configJson) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/CardigannIndexer.cs:line 987
   at Jackett.Server.Controllers.IndexerApiController.UpdateConfig(ConfigItem[] config) in /home/vsts/work/1/s/src/Jackett.Server/Controllers/IndexerApiController.cs:line 97
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/vsts/work/1/s/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 61

**Comment by ilike2burnthing** [comment]
Remove the indexer from Jackett, restart the Jackett container, sign out of Kinozal on their site, then sign back in, make sure you can access the browse page, then add Kinozal back to Jackett.

If that still fails, provide an enhanced log of the issue - https://github.com/Jackett/Jackett/wiki/Troubleshooting#how-to-provide-a-jackett-enhanced-log-for-a-ticket

**Comment by dshkoda32** [comment]
[log.txt](https://github.com/Jackett/Jackett/files/9482733/log.txt)

**Comment by ilike2burnthing** [comment]
That's not an enhanced log, please see the instructions.

**Comment by dshkoda32** [comment]
[log.txt](https://github.com/Jackett/Jackett/files/9482951/log.txt)

**Comment by ilike2burnthing** [comment]
I have zero idea what's going on with that log, I don't know if there's identifying information going on there or not (I saw at least one public IP floating around), but there's nothing to do with Kinozal that I could see in there.

Please provide an enhanced log containing only the failed login attempt for Kinozal.

**Comment by dshkoda32** [comment]
Here there is about kinozal, it feels like indexer does not accept the SID of my account
[log.txt](https://github.com/Jackett/Jackett/files/9484263/log.txt)

**Comment by garfield69** [comment]
your problem begins before the sid gets used for accessing the browse page.
when the kinozal indexer logins using the takelogin.php page with your username and password, you are being redirected to
https://kinozal-guru.appspot.com/sd_confirm
which is not expected, since after login jackett is expecting to be able to see https://kinozal-guru.appspot.com/userdetails.php

the sd_confirm page look like this:
![image](https://user-images.githubusercontent.com/6933240/188304254-ca7e94a5-5f87-488f-8818-a7bb29126048.png)


because you are using docker things are complicated.
somehow, you need to login to the website and complete the login successfully such that when you use the jackett indexer it will not be challenged.

Also, I have removed your log.txt because you did not redact your credentials.
I suggest you login to kinozal with a browser, change your password, and log out to restore your privacy.

as for jackett, @ilike2burnthing  may have some idea how to proceed, docker is not my cup of tea.

**Comment by ilike2burnthing** [comment]
See if the steps above solve your issue.

If not, please answer the questions below.

Are you using a VPN or proxy?

Are you using a VPS?

Are multiple people using your Jackett instance?

Are you using automation software which may be spamming requests to kinozal through Jackett?

**Comment by dshkoda32** [comment]
I use vps, jackett is used by several people.

**Comment by ilike2burnthing** [comment]
If Garfield's suggestion above doesn't resolve your issue then there isn't really a fix to it. Kinozal have detected your activity as suspicious, either because of the VPS IP or because it's being flooded with requests by your users.

**Comment by dshkoda32** [comment]
I put it on the vps vpn server, went from the ip server to the site, confirmed the captcha and everything worked
By the way, I also wanted to do with tapochek.net When logging in on the site, I entered a captcha, but there is still an error in jackett that I entered the code incorrectly

**Comment by ilike2burnthing** [comment]
I don't have an account, but I'm not seeing a captcha on the login page, even if I fail a login with fake details.

Please provide an HTML dump of the login page with captcha - https://github.com/Jackett/Jackett/wiki/Troubleshooting#how-to-provide-html-source

@garfield69 can you recreate it?
